### PR TITLE
[sw/testutils] Check-in autogenerated ISR testutils

### PR DESCRIPTION
**Pull request by timothytrippel**
This PR contains three commits that are described below.

Commit 1: reorganizes the core testutil autogen script logic so that it can be reused by the `util/make_new_dif.py` script. This enables the `util/make_new_dif.py` script to regenerate the autogenerated testutils when the DIFs are regenerated, as is done in CI to check for stale DIFs.

Commit 2: checks-in the autogenerated testutil code, in line with the convention for checking in autogenerated DIFs.

Commit 3: updates the `util/make_new_dif.py` script to autogenerate the testutils after updating autogenerating DIFs that are invoked by said testutils. An **_intended_** side effect of this commit is that autogenerated testutils functions will be checked for staleness in CI, [when the autogenerated DIFs are checked](https://github.com/lowRISC/opentitan/blob/4b470bc12e5da64e6c9c7eed538962d71cd698a4/ci/scripts/check-generated.sh#L66).

This PR addresses several tasks in #9038.

Note: this PR is fairly large due to the inclusion of autogenerated code.

**_Note: This PR depends on #11028._**

### CODE REVIEW

File: util/autogen_testutils/gen.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    # Create output directories if needed.
+    autogen_testutils_dir.mkdir(exist_ok=True)
+
+    # Render templates.
+    for suffix in [".h", ".c", ".build"]:
+        comment_syntax = "//"
+        if suffix == ".build":
+            comment_syntax = "#"
+        for testutils_template_path_str in glob.iglob(
+                str(testutils_templates_dir / f"*{suffix}.tpl")):
```
Comment (mcy) [inline util/autogen_testutils/gen.py:50]: I think iterdir + `if suffix == .tpl: continue` is cleaner.

File: util/autogen_testutils/gen.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            testutils_template_path = Path(testutils_template_path_str)
+
+            # Read in template, render it, and write it to the output file.
+            testutils_template = Template(testutils_template_path.read_text())
+            testutils = Path(autogen_testutils_dir /
```
Comment (mcy) [inline util/autogen_testutils/gen.py:55]: This Path() constructor seems redundant?

File: util/autogen_testutils/gen.py (hunk 1 above)
Comment (timothytrippel) [reply util/autogen_testutils/gen.py]: Fixed.

File: util/autogen_testutils/gen.py (hunk 2 above)
Comment (timothytrippel) [reply util/autogen_testutils/gen.py]: Oops forgot to clean that up. Fixed.

**Review by mcy — APPROVED** [review]

**Comment by timothytrippel** [comment]
PR dependency removed. Ready to merge when CI passes.

**Pull request merged**